### DISCUSSION TITLE
Whitelisting or blacklisting email provider during registration

### DISCUSSION THREAD

**Discussion by jaypeemayo** [OrchardCMS/OrchardCore#10918] [Q&A]
Hi I only want gmail and yahoo emails when user is registering to our website..is there a way to whitelist them and  dont allow the rest?

**Comment by ns8482e** [comment]
May be you just don't use ootb registration - instead, crete your own controller and disable ootb from site settings.

**Comment by hishamco** [comment]
You can use what @ns8482e suggested before or this can be doable using events, if we have `UserCreate` event you can deny the emails easly. Let me check the available user events

**Reply by hishamco** [reply]
I just found there's `IRegistrationFormEvents`, I think you need a custom event handler

**Reply by ns8482e** [reply]
@jaypeemayo you can add custom validation errors implementing `IRegistrationFormEvents`
